src/utility/connector.cpp

```diff
@@ -166,22 +166,31 @@ void connector::handle_resolve(const boost_code& ec, asio::iterator iterator,
         return;
     }
 
-    const auto timeout = settings_.connect_timeout();
-    const auto timer = std::make_shared<deadline>(pool_, timeout);
-    const auto socket = std::make_shared<network::socket>(pool_);
-
-    // Retain a socket reference until connected, allowing connect cancelation.
-    pending_.store(socket);
-
-    // Manage the socket-timer race, terminating if either fails.
-    const auto handle_connect = synchronize(handler, 1, NAME, mode);
-
-    // This is branch #1 of the connnect sequence.
-    timer->start(
-        std::bind(&connector::handle_timer,
-            shared_from_this(), _1, socket, handle_connect));
-
-    safe_connect(iterator, socket, timer, handle_connect);
+    auto do_connecting = [&](boost::asio::ip::tcp::resolver::iterator iter){
+		const auto timeout = settings_.connect_timeout();
+		const auto timer = std::make_shared<deadline>(pool_, timeout);
+		const auto socket = std::make_shared<network::socket>(pool_);
+
+		// Retain a socket reference until connected, allowing connect cancelation.
+		pending_.store(socket);
+
+		// Manage the socket-timer race, terminating if either fails.
+		const auto handle_connect = synchronize(handler, 1, NAME, mode);
+
+		// This is branch #1 of the connnect sequence.
+		timer->start(
+			std::bind(&connector::handle_timer,
+				shared_from_this(), _1, socket, handle_connect));
+
+		safe_connect(iter, socket, timer, handle_connect);
+    };
+
+	//get all hosts under one seed
```

Comment:
Generalize comments ("DNS record" vs. "seed"), since this affects all outbound network connections, not just seeding.

Style: use sentences (leading space, leading cap, period) in comments, e.g.:
```cpp
// Get all hosts under one DNS record.
```